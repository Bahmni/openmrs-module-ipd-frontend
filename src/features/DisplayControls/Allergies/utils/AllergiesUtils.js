import axios from "axios";
import { ALLERGIES_BASE_URL } from "../../../../constants";

export const fetchAllergiesData = async (patientUuid) => {
  const FETCH_ALLERGIES_URL = `${ALLERGIES_BASE_URL}?patient=${patientUuid}&_summary=data`;
  try {
    const response = await axios.get(FETCH_ALLERGIES_URL);
    return response;
  } catch (error) {
    return error;
  }
};
