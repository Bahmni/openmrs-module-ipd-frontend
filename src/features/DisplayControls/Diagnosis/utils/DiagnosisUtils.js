import axios from "axios";
import { DIAGNOSIS_SEARCH_URL } from "../../../../constants";

export const getPatientDiagnosis = async (patientUuid) => {
  try {
    const response = await axios.get(DIAGNOSIS_SEARCH_URL, {
      params: { patientUuid: patientUuid },
      withCredentials: true,
    });
    return response.data;
  } catch (error) {
    return error;
  }
};

export const diagnosisHeaders = [
  {
    id: "1",
    header: "Diagnosis",
    key: "diagnosis",
    isSortable: false,
  },
  {
    id: "2",
    header: "Order",
    key: "order",
    isSortable: true,
  },
  {
    id: "3",
    header: "Certainty",
    key: "certainty",
    isSortable: true,
  },
  {
    id: "4",
    header: "Status",
    key: "status",
    isSortable: true,
  },
  {
    id: "5",
    header: "Diagnosed By",
    key: "diagnosedBy",
    isSortable: false,
  },
  {
    id: "6",
    header: "Diagnosis Date ",
    key: "diagnosisDate",
    isSortable: true,
  },
];
