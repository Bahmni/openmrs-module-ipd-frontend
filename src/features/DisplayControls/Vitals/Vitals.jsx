import React from "react";

const Vitals = () => {
  return <div>Vitals</div>;
};

export default Vitals;
