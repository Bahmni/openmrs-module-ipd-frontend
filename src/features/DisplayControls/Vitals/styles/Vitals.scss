.no-vitals-message {
  padding-top: 10px;
}

.vital-table {
  overflow: visible;
  overflow-x: auto;
  padding: 10px 0;

  .vital-date-time {
    display: flex;
    flex-flow: row;
    justify-content: flex-start;
    align-items: baseline;
    margin-left: 10px;
    a {
      margin-left: 10px;
    }
  }

  .vitals-biometrics-history {
    display: flex;
    flex-flow: column;
    padding: 0;
    border-top: 1px solid #e5e5e5;
  }

  .vital-tiles,
  .abnormal-tiles {
    margin: 3px;
    padding: 10px;
    display: flex;
    flex-flow: column;
    align-items: normal;
    text-align: left;
    background-color: whitesmoke;
    justify-content: space-between;
    min-width: fit-content;
  }

  .vital-values {
    font-size: 16px;
    min-width: max-content;
  }
  .vital-headers {
    font-size: 12px;
  }
  .vital-units {
    font-size: 14px;
  }
  .vital-value-and-unit {
    display: flex;
    flex-flow: row;
    justify-content: space-between;
  }

  .abnormal-tiles {
    color: #da1e28;
  }

  .vitals-history-pagination,
  .biometrics-history-pagination {
    width: 100%;
    border: 1px solid #e0e0e0;
  }

  .vitals-history-title,
  .biometrics-history-title {
    padding: 10px;
    border: 1px solid #e0e0e0;
    background-color: #fff;
    font-weight: 600;
    font-size: 15px;
  }
  .history-abnormal-tile{
    color: #da1e28;
  }
  .history-tile{
    color: #525252;
  }
  .history-data-table{
      pointer-events: none;
  }
}
