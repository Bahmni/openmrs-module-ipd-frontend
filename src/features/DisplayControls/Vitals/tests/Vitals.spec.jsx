import { render, screen, waitFor, fireEvent, act } from "@testing-library/react";
import React from "react";
import Vitals from "../components/Vitals";
import { getPatientVitals,getPatientVitalsHistory } from "../utils/VitalsUtils";

jest.mock("../utils/VitalsUtils", () => {
  const originalModule = jest.requireActual("../utils/VitalsUtils");
  return {
    ...originalModule,
    getPatientVitals: jest.fn(),
    getPatientVitalsHistory: jest.fn()
  };
});

describe("Vitals", () => {
  const mockVitalsData = {
    identifiers: [
      {
        identifier: "12345",
      },
    ],
    conceptDetails: [
      {
        name: "WEIGHT",
        fullName: "WEIGHT",
        units: "Kg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "BMI",
        fullName: "BMI Data",
        units: null,
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Diastolic Blood Pressure",
        fullName: "Diastolic Blood Pressure",
        units: "mmHg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Systolic Blood Pressure",
        fullName: "Systolic Blood Pressure",
        units: "mmHg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "SpO2",
        fullName: "Arterial Blood Oxygen Saturation (Pulse Oximeter)",
        units: "%",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Pulse",
        fullName: "Pulse",
        units: "beats/min",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Temperature",
        fullName: "Temperature",
        units: "C",
        hiNormal: 37.7,
        lowNormal: 35,
        attributes: {},
      },
      {
        name: "Respiratory Rate",
        fullName: "Respiratory Rate",
        units: null,
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
    ],
    tabularData: {
      "2023-11-07T16:02:11+05:30": {
        "Diastolic Blood Pressure": {
          value: "80.0",
          abnormal: false,
        },
        "Systolic Blood Pressure": {
          value: "110.0",
          abnormal: false,
        },
        Pulse: {
          value: "90.0",
          abnormal: false,
        },
        Temperature: {
          value: "36.0",
          abnormal: false,
        },
        "Respiratory Rate": {
          value: "25.0",
          abnormal: false,
        },
        SpO2: {
          value: "93.0",
          abnormal: false,
        },
        BMI: {
          value: "19.53",
          abnormal: false,
        },
        WEIGHT: {
          value: "50.0",
          abnormal: false,
        },
        HEIGHT: {
          value: "160.0",
          abnormal: false,
        },
      },
    },
  };
   const mockVitalsHistoryData = {
    identifiers: [
      {
        identifier: "12345",
      },
    ],
    conceptDetails: [
      {
        name: "WEIGHT",
        fullName: "WEIGHT",
        units: "Kg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "BMI",
        fullName: "BMI Data",
        units: null,
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Diastolic Blood Pressure",
        fullName: "Diastolic Blood Pressure",
        units: "mmHg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Systolic Blood Pressure",
        fullName: "Systolic Blood Pressure",
        units: "mmHg",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "SpO2",
        fullName: "Arterial Blood Oxygen Saturation (Pulse Oximeter)",
        units: "%",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Pulse",
        fullName: "Pulse",
        units: "beats/min",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "Temperature",
        fullName: "Temperature",
        units: "C",
        hiNormal: 37.7,
        lowNormal: 35,
        attributes: {},
      },
      {
        name: "Respiratory Rate",
        fullName: "Respiratory Rate",
        units: null,
        hiNormal: null,
        lowNormal: null,
        attributes: {},
      },
      {
        name: "MUAC",
        fullName: "Mid-upper arm circumference",
        units: "Cm",
        hiNormal: null,
        lowNormal: null,
        attributes: {},
    }
    ],
    tabularData: {
      "2023-11-07T16:02:11+05:30": {
        "Diastolic Blood Pressure": {
          value: "80.0",
          abnormal: false,
        },
        "Systolic Blood Pressure": {
          value: "110.0",
          abnormal: false,
        },
        Pulse: {
          value: "90.0",
          abnormal: false,
        },
        Temperature: {
          value: "36.0",
          abnormal: false,
        },
        "Respiratory Rate": {
          value: "25.0",
          abnormal: false,
        },
        SpO2: {
          value: "93.0",
          abnormal: false,
        },
        BMI: {
          value: "19.53",
          abnormal: false,
        },
        WEIGHT: {
          value: "50.0",
          abnormal: false,
        },
        HEIGHT: {
          value: "160.0",
          abnormal: false,
        },
        MUAC: {
          value: "26.0",
          abnormal: false
      }
      },
      "2023-12-11T19:32:32+05:30": {
        Pulse: {
            value: "100.0",
            abnormal: false
        },
        "Diastolic Blood Pressure": {
            value: "84.0",
            abnormal: false
        },
        "Systolic Blood Pressure": {
           value: "115.0",
            abnormal: false
        },
        "Respiratory Rate": {
            value: "20.0",
            abnormal: false
        },
        SpO2: {
            value: "94.0",
            abnormal: false
        },
        Temperature: {
            value: "37.0",
            abnormal: false
        },
        BMI: {
            value: "19.0",
            abnormal: false
        },
        HEIGHT: {
            value: "165.0",
            abnormal: false
        },
        WEIGHT: {
            value: "50.0",
            abnormal: false
        },
        MUAC: {
            value: "27.0",
            abnormal: false
        }
    },
    }
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("displays vital and biometrics history after data is fetched", async () => {
    getPatientVitals.mockResolvedValueOnce(mockVitalsData);
    getPatientVitalsHistory.mockResolvedValueOnce(mockVitalsHistoryData);
    const {container} = render(<Vitals patientId="123" />);
      
    await waitFor(() => {
          expect(screen.getByText(/Vitals History/i)).toBeTruthy();
       });
    const vitalsHistory = screen.getByText(/Vitals History/i);  
    act(() => {
        fireEvent.click(vitalsHistory);
      });  
    await waitFor(() => {
      expect(screen.getByText("Pulse (beats/min)")).toBeTruthy();
      expect(screen.getByText(/Nutritional Values/i)).toBeTruthy();
      expect(container).toMatchSnapshot();
       });
  });

  it("should display vital details after data is fetched", async () => {
    getPatientVitals.mockResolvedValueOnce(mockVitalsData);
    getPatientVitalsHistory.mockResolvedValueOnce(mockVitalsHistoryData);
    const {container} = render(<Vitals patientId="123" />);
    await waitFor(() => {
        expect(screen.getByText(/Temp/i)).toBeTruthy();
        expect(container).toMatchSnapshot();

    });
  });

  it("should display text when vitals is not present", async () => {
    getPatientVitalsHistory.mockResolvedValueOnce(mockVitalsHistoryData);
    getPatientVitals.mockResolvedValueOnce({
      identifiers: [
        {
          identifier: "12345",
        },
      ],
      conceptDetails: [
        {
          name: "WEIGHT",
          fullName: "WEIGHT",
          units: "Kg",
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "BMI",
          fullName: "BMI Data",
          units: null,
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "Diastolic Blood Pressure",
          fullName: "Diastolic Blood Pressure",
          units: "mmHg",
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "Systolic Blood Pressure",
          fullName: "Systolic Blood Pressure",
          units: "mmHg",
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "SpO2",
          fullName: "Arterial Blood Oxygen Saturation (Pulse Oximeter)",
          units: "%",
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "Pulse",
          fullName: "Pulse",
          units: "beats/min",
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
        {
          name: "Temperature",
          fullName: "Temperature",
          units: "C",
          hiNormal: 37.7,
          lowNormal: 35,
          attributes: {},
        },
        {
          name: "Respiratory Rate",
          fullName: "Respiratory Rate",
          units: null,
          hiNormal: null,
          lowNormal: null,
          attributes: {},
        },
      ],
      tabularData: {},
    });
    render(<Vitals patientId="123" />);
    await waitFor(() =>
      expect(
        screen.getByText("No Vitals available for this patient")
      ).toBeTruthy()
    );
  });

  it("renders without crashing", () => {
    getPatientVitals.mockResolvedValueOnce(mockVitalsData);
    render(<Vitals patientId="123" />);
  });

  it("calls getPatientVitals on mount", () => {
    getPatientVitals.mockResolvedValueOnce(mockVitalsData);
    render(<Vitals patientId="123" />);
    expect(getPatientVitals).toHaveBeenCalledWith("123");
  });

  it("displays loading skeleton while fetching data", () => {
    getPatientVitals.mockResolvedValueOnce(mockVitalsData);
    render(<Vitals patientId="123" />);
    expect(screen.getByTestId("header-loading")).toBeTruthy();
  });

});


