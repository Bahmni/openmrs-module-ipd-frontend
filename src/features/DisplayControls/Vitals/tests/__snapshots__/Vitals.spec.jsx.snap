// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vitals displays vital and biometrics history after data is fetched 1`] = `
<div>
  <div
    class="bx--tile vital-table"
  >
    <div
      class="vital-date-time"
    >
      07/11/2023
      , 10:32 AM  
      <a
        class="bx--link show-more"
        kind="tertiary"
      >
        Vitals History
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 10L26 20 24.6 21.4 16 12.8 7.4 21.4 6 20z"
          />
        </svg>
      </a>
    </div>
    <div
      class="bx--row"
    >
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Pulse
          </span>
          <span
            class="vital-values"
          >
            90
             
            <span
              class="vital-units"
            >
              beats/min
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vitsal-headers"
          >
            SpO2
          </span>
          <span
            class="vital-values"
          >
            93
             
            <span
              class="vital-units"
            >
              %
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            R.rate
          </span>
          <span
            class="vital-values"
          >
            25
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Temp
          </span>
          <span
            class="vital-values"
          >
            36.0
             
            <span
              class="vital-units"
            >
              C
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            BP
          </span>
          <span
            class="vital-values"
          >
            110/80
             
            <span
              class="vital-units"
            >
              mmHg
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Height
          </span>
          <span
            class="vital-values"
          >
            160
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Weight
          </span>
          <span
            class="vital-values"
          >
            50
             
            <span
              class="vital-units"
            >
              Kg
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            BMI
          </span>
          <span
            class="vital-values"
          >
            19.53
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
    </div>
    <div
      class="bx--tile vitals-biometrics-history"
    >
      <div
        class="bx--tile"
      >
        <div
          class="vitals-history-title"
        >
          Vitals
        </div>
        <div
          class="bx--data-table-content"
        >
          <table
            class="bx--data-table history-data-table bx--data-table--normal bx--data-table--zebra bx--data-table--no-border"
            data-testid="vitals-datatable"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Date and Time
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Pulse (beats/min)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    SPO2 (%)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    R.rate (breaths/min)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Temp (DEG C)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    BP (mmHg)
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              aria-live="polite"
            >
              <tr>
                <td>
                  07/11/2023, 10:32 AM
                </td>
                <td
                  class="history-tile"
                >
                  90.0
                </td>
                <td
                  class="history-tile"
                >
                  93.0
                </td>
                <td
                  class="history-tile"
                >
                  25.0
                </td>
                <td
                  class="history-tile"
                >
                  36.0
                </td>
                <td
                  class="history-tile"
                >
                  110/80
                </td>
              </tr>
              <tr>
                <td>
                  11/12/2023, 02:02 PM
                </td>
                <td
                  class="history-tile"
                >
                  100.0
                </td>
                <td
                  class="history-tile"
                >
                  94.0
                </td>
                <td
                  class="history-tile"
                >
                  20.0
                </td>
                <td
                  class="history-tile"
                >
                  37.0
                </td>
                <td
                  class="history-tile"
                >
                  115/84
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="bx--pagination vitals-history-pagination bx--pagination--md"
        >
          <div
            class="bx--pagination__left"
          >
            <label
              class="bx--pagination__text"
              for="bx-pagination-select-1"
              id="bx-pagination-select-1-count-label"
            >
              Items per page:
            </label>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__item-count"
              >
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-1"
                    >
                      <option
                        class="bx--select-option"
                        value="5"
                      >
                        5
                      </option>
                      <option
                        class="bx--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="bx--select-option"
                        value="20"
                      >
                        20
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="bx--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text bx--pagination__items-count"
            >
              1–2 of 2 items
            </span>
          </div>
          <div
            class="bx--pagination__right"
          >
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__page-number"
              >
                <label
                  class="bx--label bx--visually-hidden"
                  for="bx-pagination-select-1-right"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-1-right"
                    >
                      <option
                        class="bx--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="bx--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text"
            >
              of 1 page
            </span>
            <div
              class="bx--pagination__control-buttons"
            >
              <button
                class="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
                disabled=""
                tabindex="0"
                type="button"
              >
                <div
                  class="bx--assistive-text"
                >
                  Previous page
                </div>
                <svg
                  aria-hidden="true"
                  aria-label="Previous page"
                  class="bx--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 24L10 16 20 8z"
                  />
                </svg>
              </button>
              <button
                class="bx--pagination__button bx--pagination__button--forward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-end"
                disabled=""
                tabindex="0"
                type="button"
              >
                <div
                  class="bx--assistive-text"
                >
                  Next page
                </div>
                <svg
                  aria-hidden="true"
                  aria-label="Next page"
                  class="bx--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8L22 16 12 24z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bx--tile"
      >
        <div
          class="biometrics-history-title"
        >
          Nutritional Values
        </div>
        <div
          class="bx--data-table-content"
        >
          <table
            class="bx--data-table history-data-table bx--data-table--normal bx--data-table--zebra bx--data-table--no-border"
            data-testid="biometrics-datatable"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Date and Time
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Height (cm)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    Weight (kg)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    BMI (kg/m2)
                  </div>
                </th>
                <th
                  scope="col"
                >
                  <div
                    class="bx--table-header-label"
                  >
                    MUAC (cm)
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              aria-live="polite"
            >
              <tr>
                <td>
                  07/11/2023, 10:32 AM
                </td>
                <td
                  class="history-tile"
                >
                  160.0
                </td>
                <td
                  class="history-tile"
                >
                  50.0
                </td>
                <td
                  class="history-tile"
                >
                  19.53
                </td>
                <td
                  class="history-tile"
                >
                  26.0
                </td>
              </tr>
              <tr>
                <td>
                  11/12/2023, 02:02 PM
                </td>
                <td
                  class="history-tile"
                >
                  165.0
                </td>
                <td
                  class="history-tile"
                >
                  50.0
                </td>
                <td
                  class="history-tile"
                >
                  19.0
                </td>
                <td
                  class="history-tile"
                >
                  27.0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="bx--pagination biometrics-history-pagination bx--pagination--md"
        >
          <div
            class="bx--pagination__left"
          >
            <label
              class="bx--pagination__text"
              for="bx-pagination-select-2"
              id="bx-pagination-select-2-count-label"
            >
              Items per page:
            </label>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__item-count"
              >
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-2"
                    >
                      <option
                        class="bx--select-option"
                        value="5"
                      >
                        5
                      </option>
                      <option
                        class="bx--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="bx--select-option"
                        value="20"
                      >
                        20
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="bx--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text bx--pagination__items-count"
            >
              1–2 of 2 items
            </span>
          </div>
          <div
            class="bx--pagination__right"
          >
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__page-number"
              >
                <label
                  class="bx--label bx--visually-hidden"
                  for="bx-pagination-select-2-right"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-2-right"
                    >
                      <option
                        class="bx--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="bx--select__arrow"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text"
            >
              of 1 page
            </span>
            <div
              class="bx--pagination__control-buttons"
            >
              <button
                class="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-center"
                disabled=""
                tabindex="0"
                type="button"
              >
                <div
                  class="bx--assistive-text"
                >
                  Previous page
                </div>
                <svg
                  aria-hidden="true"
                  aria-label="Previous page"
                  class="bx--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 24L10 16 20 8z"
                  />
                </svg>
              </button>
              <button
                class="bx--pagination__button bx--pagination__button--forward bx--pagination__button--no-index bx--btn bx--btn--ghost bx--btn--disabled bx--tooltip--hidden bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--btn--icon-only--top bx--tooltip--align-end"
                disabled=""
                tabindex="0"
                type="button"
              >
                <div
                  class="bx--assistive-text"
                >
                  Next page
                </div>
                <svg
                  aria-hidden="true"
                  aria-label="Next page"
                  class="bx--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8L22 16 12 24z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Vitals should display vital details after data is fetched 1`] = `
<div>
  <div
    class="bx--tile vital-table"
  >
    <div
      class="vital-date-time"
    >
      07/11/2023
      , 10:32 AM  
      <a
        class="bx--link show-more"
        kind="tertiary"
      >
        Vitals History
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 22L6 12 7.4 10.6 16 19.2 24.6 10.6 26 12z"
          />
        </svg>
      </a>
    </div>
    <div
      class="bx--row"
    >
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Pulse
          </span>
          <span
            class="vital-values"
          >
            90
             
            <span
              class="vital-units"
            >
              beats/min
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vitsal-headers"
          >
            SpO2
          </span>
          <span
            class="vital-values"
          >
            93
             
            <span
              class="vital-units"
            >
              %
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            R.rate
          </span>
          <span
            class="vital-values"
          >
            25
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Temp
          </span>
          <span
            class="vital-values"
          >
            36.0
             
            <span
              class="vital-units"
            >
              C
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            BP
          </span>
          <span
            class="vital-values"
          >
            110/80
             
            <span
              class="vital-units"
            >
              mmHg
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Height
          </span>
          <span
            class="vital-values"
          >
            160
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            Weight
          </span>
          <span
            class="vital-values"
          >
            50
             
            <span
              class="vital-units"
            >
              Kg
            </span>
             
          </span>
        </div>
      </div>
      <div
        class="bx--col"
      >
        <div
          class="bx--tile vital-tiles"
        >
          <span
            class="vital-headers"
          >
            BMI
          </span>
          <span
            class="vital-values"
          >
            19.53
             
            <span
              class="vital-units"
            />
             
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
